integration-tests/tests/test_integration.py

```diff
@@ -516,12 +534,85 @@ async def test_update_price_simple_with_keypair_hotload(self, client_hotload: Py
         # Continue normally with the existing simple scenario
         await self.test_update_price_simple(client_hotload)
 
+    @pytest.mark.asyncio
+    async def test_update_price_discards_unpermissioned(self, client: (PythAgentClient, str)):
+
+        (client, tmp_path) = client
+
+        # Fetch all products
+        products = {product["attr_dict"]["symbol"]: product for product in await client.get_all_products()}
+
+        # Find the product account ID corresponding to the BTC/USD symbol
+        product = products[BTC_USD["attr_dict"]["symbol"]]
+        product_account = product["account"]
+
+        # Get the price account with which to send updates
+        price_account = product["price_accounts"][0]["account"]
+
+        # Use the unpermissioned ETH/USD symbol to trigger unpermissioned account filtering
+        product_unperm = products[ETH_USD["attr_dict"]["symbol"]]
+        product_account_unperm = product_unperm["account"]
+        price_account_unperm = product_unperm["price_accounts"][0]["account"]
+
+        # Send an "update_price" request for the valid symbol
+        await client.update_price(price_account, 42, 2, "trading")
+        time.sleep(1)
+
+        # Send another "update_price" request to trigger aggregation
+        await client.update_price(price_account, 81, 1, "trading")
+        time.sleep(2)
+
+        # Send an "update_price" request for the invalid symbol
+        await client.update_price(price_account_unperm, 48, 2, "trading")
+        time.sleep(1)
+
+        # Send another "update_price" request to "trigger" aggregation
+        await client.update_price(price_account_unperm, 81, 1, "trading")
+        time.sleep(2)
+
+        # Confirm that the valid price account has been updated with the values from the first "update_price" request
+        final_product_state = await client.get_product(product_account)
+
+        final_price_account = final_product_state["price_accounts"][0]
+        assert final_price_account["price"] == 42
+        assert final_price_account["conf"] == 2
+        assert final_price_account["status"] == "trading"
+
+        # Sanity-check that the unpermissioned symbol was not updated
+        final_product_state_unperm = await client.get_product(product_account_unperm)
+
+        final_price_account_unperm = final_product_state_unperm["price_accounts"][0]
+        assert final_price_account_unperm["price"] == 0
+        assert final_price_account_unperm["conf"] == 0
+        assert final_price_account_unperm["status"] == "unknown"
+
+        # Confirm agent logs contain the relevant WARN log
+        with open(f"{tmp_path}/agent_logs/stdout") as f:
+            contents = f.read()
+            lines_found = 0
+            for line in contents.splitlines():
```

Comment:
nitpik: you can do `for line in f` to be more efficient

Reply:
I was hesitant to do that because while this assertion takes place, the agent is still running and appending to the file which could create an infinite loop.

Reply:
oh smart, you are right.